### PR TITLE
feat(VPagination): update to v3

### LINKED ISSUE #2070: [Feature Request] Add min to pagination
- What will it allow you to do that you can't do today?

> I want to use pagination for switching between years. So the pagination here does not start with "1". Instead it starts for example at 2010 (this of course varies).

- How will it make current work-arounds straightforward?

> I do no longer need to manually catch if the user selects years (or pages) I do not want.

- What potential bugs and edge cases does it help to avoid?

> Manual catching. I think this is enough.

---

The whole thing does not depend on years. It is just a convenient way to prevent user select information which is not available or something. I think `max` is already handled by the prop `length`. But there is no way to define some property like `min`.

### LINKED ISSUE #2541: [Feature Request] Pagination flat/outline styles
### New Functionality
Pagination can't be styled out of the box as flat or outline buttons

### Improvements
you need to create your own styles to achieve the desired effect that buttons have by default

### LINKED ISSUE #4201: [Bug Report] Pagination-Buttons have outline
### Versions and Environment
**Vuetify:** 1.1.0-beta.1
**Vue:** 2.5.16
**Browsers:** Chrome 67.0.3396.62
**OS:** Windows 7

### Steps to reproduce
Use pagination with `round` attribute.

### Expected Behavior
No outline.

### Actual Behavior
There is an outline.

### Reproduction Link
[https://codepen.io/anon/pen/Vdazxp](https://codepen.io/anon/pen/Vdazxp)

### Other comments
Can be fixed using `button { outline: none; }`.

### LINKED ISSUE #5028: [Feature Request] add a11y support to v-pagination
### What problem does this feature solve?
Adding keyboard navigation to the component will improve Accessibility usage; e.g. people with disabilities who primarily rely on keyboard navigation would greatly benefit from it.

### What is your proposed solution?
Simple use case:

template:
```
<v-pagination
        v-model="page"
        :length="numPages"
        @keypress = "keyHandler"
/>
```

script:
```
keyHandler(btn) {
  switch(btn.key) {
     case 'ArrowRight': /* navigate to the next page */
     break;
     case 'ArrowLeft': /*navigate to previous page*/
     break;
  }
}
```

Perhaps this has been implemented in some other form I'm unaware of?

### LINKED ISSUE #8146: [Bug Report] Paginations are still breaking out of their parent component
### Environment
**Vuetify Version:** 2.0.2
**Vue Version:** 2.6.10
**Browsers:** Safari 12.1.1
**OS:** Mac OS 10.14.5

### Steps to reproduce
Open the codepen.

### Expected Behavior
The entire pagination component should fit within the parent card component.

### Actual Behavior
Both the prior and next buttons are extending outside the parent card component.

### Reproduction Link
<a href="https://codepen.io/anon/pen/PMNzLj?&editable=true&editors=101" target="_blank">https://codepen.io/anon/pen/PMNzLj?&editable=true&editors=101</a>

### Other comments
Changing the width of the card to 500 works while 600 does not.
The v2.0.2 changes made for #7947 didn't completely fix the previously reported bug..

### LINKED ISSUE #8759: [Feature Request] <a> tag for pagination
### Problem to solve
For SEO purpose, a tag is very import to the html content. Now v-pagination uses button, i suggest an alternative choice to use a tag.

### Proposed solution
Add some new attributes to the pagination component like, tag, to, nuxt, link or something else.

### LINKED ISSUE #10085: [Feature Request] comma separating v-pagination numbers
### Environment
**Vuetify Version:** 2.1.15
**Vue Version:** 3.0.0-alpha.0
**Browsers:** Firefox 71.0
**OS:** Mac OS 10.15

### Steps to reproduce
set length prop of v-pagination to > 1000 . 

### Expected Behavior
more human readable number display

### Actual Behavior
no ./, delimiters 

### Reproduction Link
<a href="https://codepen.io/hp4k1h5/pen/ExawwJd" target="_blank">https://codepen.io/hp4k1h5/pen/ExawwJd</a>

### Other comments
i think it can be solved in line 173 of https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/components/VPagination/VPagination.ts#L173
by changing it to 
```      }), [i.toLocaleString()])```

### LINKED ISSUE #11242: [Bug Report] v-pagination component too wide, if length is large
### Environment
**Vuetify Version:** 2.2.23
**Vue Version:** 2.6.11
**Browsers:** Chrome 81.0.4044.92
**OS:** Windows 10

### Steps to reproduce
Create a v-pagination-component with large length, i.e.

      <v-pagination
        v-model="page"
        :length="10000"
      ></v-pagination>

### Expected Behavior
The rendered component should fit into the available screen size.

### Actual Behavior
The component is larger than the available screen size.

### Reproduction Link
<a href="https://codepen.io/laberning/pen/rNOmbdL" target="_blank">https://codepen.io/laberning/pen/rNOmbdL</a>

### Other comments
The implementation assumes a fixed width per "page"-button, but pages with high numbers need more space.

this.maxButtons = Math.floor((width - 96) / 42)

### LINKED ISSUE #11662: [Bug Report] Bug v-pagination mobile phone
### Environment
**Vuetify Version:** 2.3.1
**Vue Version:** 2.6.1
**Browsers:** Chrome 83.0.4103.97
**OS:** Windows 10

### Steps to reproduce
Create pagination element with many pages.
If total-visible is less than 5, the buttons of the current page are not visible

### Expected Behavior
The current page button is visible

### Actual Behavior
buttons of the current page are not visible

### Reproduction Link
<a href="https://codepen.io/OlegF/pen/jOWVyQa" target="_blank">https://codepen.io/OlegF/pen/jOWVyQa</a>

### Other comments
If total-visible is less than 5, the buttons of the current page are not visible
The current page button is visible

The problem is also relevant for mobile phones when the window width is less than 320 pixels.

### LINKED ISSUE #12232: [Bug Report] Pagination show two times ... in between start and end page numbers
### Environment
**Vuetify Version:** 2.3.10
**Vue Version:** 3.0.0-rc.10
**Browsers:** Chrome 85.0.4183.102
**OS:** Windows 8.1

### Steps to reproduce
please click next button

### Expected Behavior
https://ibb.co/Vwcxqt7
i feel it should be single "..." or inbetween some page number like following  [1][2]...[5]....[9] instead of [1][2]... ....[9] 

### Actual Behavior
Actual showing like following 
[1][2]... ....[9] 

### Reproduction Link
<a href="https://codepen.io/sujithfem/pen/yLOqJXX?editable=true&editors=101%3Dhttps%3A%2F%2Fvuetifyjs.com%2Fen%2Fcomponents%2Fpaginations%2F" target="_blank">https://codepen.io/sujithfem/pen/yLOqJXX?editable=true&editors=101%3Dhttps%3A%2F%2Fvuetifyjs.com%2Fen%2Fcomponents%2Fpaginations%2F</a>

### LINKED ISSUE #12395: [Feature Request] v-pagination different visibilty
### Problem to solve
v-pagination provides this template:
< [ 1 ] [ 2 ] [ 3 ] ... [ n -2 ] [ n-1 ] [ n ] > 
What i want is:
< [ 1 ] [ 2 ] [ 3 ] [ 4 ] [ 5 ] ... > or < ... [ k-1 ] [ k ] [ k+1 ] ... >

### PR DESCRIPTION
**Pull request by nekosaur**
## Description

update v-pagination to v3

resolves #4201
fixes #11242
resolves #8759 (use slot)
fixes #12395
resolves #2070
resolves #5028
fixes #8146
resolves #2541
fixes #11662
fixes #12232
resolves #10085

## Breaking changes / additions
- v-model updated to `model-value`/`update:model-value`
- removed `circle` prop
- removed `light` and `dark` props
- renamed `wrapper-aria-label` to `aria-label`
- added `first-icon`
- added `last-icon`
- added `first-aria-label`
- added `last-aria-label`
- added `show-first-last-page`
- added `ellipsis` prop
- added `start` prop
- added `first`, `prev`, `next`, `last`, `item` slots

## Motivation and Context

v3

## How Has This Been Tested?

playground and unit tests

## Markup:

<details>

```vue
<template>
  <v-app class="fill-height">
    <input type="checkbox" :checked="current === 'dark'" @change="current = current === 'dark' ? 'light' : 'dark'"> theme
    <input type="checkbox" :checked="rtl" @change="rtl = !rtl"> rtl
    <div class="pa-10">
      <v-locale-provider :rtl="rtl">
        <hr>
        rounded
        <v-pagination :length="12" v-model="page" rounded="0" />
        <v-pagination :length="12" v-model="page" />
        <v-pagination :length="12" v-model="page" rounded />
        <hr>
        density
        <v-pagination :length="12" v-model="page" density="compact" />
        <v-pagination :length="12" v-model="page" />
        <v-pagination :length="12" v-model="page" density="default" />
        <hr>
        size
        <v-pagination :length="12" v-model="page" size="x-small" />
        <v-pagination :length="12" v-model="page" />
        <v-pagination :length="12" v-model="page" size="x-large" />
        <hr>
        elevation
        <v-pagination :length="12" v-model="page" elevation="0" />
        <v-pagination :length="12" v-model="page" />
        <v-pagination :length="12" v-model="page" elevation="4" />
        <hr>
        border
        <v-pagination :length="12" v-model="page" outlined elevation="0" />
        <v-pagination :length="12" v-model="page" elevation="0" />
        <v-pagination :length="12" v-model="page" border elevation="0" />
        <hr>
        show-first-last-page
        <v-pagination :length="12" v-model="page" />
        <v-pagination :length="12" v-model="page" show-first-last-page />
        <hr>
        disabled
        <v-pagination :length="12" v-model="page" />
        <v-pagination :length="12" v-model="page" disabled />
        <hr>
        total-visible
        <v-pagination :length="12" v-model="page" total-visible="7" />
        <v-pagination :length="12" v-model="page" total-visible="8" />
        <v-pagination :length="12" v-model="page" total-visible="9" />
        <v-pagination :length="12" v-model="page" total-visible="10" />
        <hr>
        resize observer
        <div style="width: 500px">
          <v-pagination :length="12" v-model="page" />
        </div>
        <hr>
        start
        <v-pagination :length="10" :start="15" show-first-last-page />
        <hr>
        loooong
        <v-pagination :length="1000" />
        <hr>
        {{ page }}
      </v-locale-provider>
    </div>
  </v-app>
</template>

<script>
  import { ref } from 'vue'
  import { useTheme } from '../src/composables/theme'

  export default {
    name: 'Playground',
    setup () {
      const show = ref(false)
      const theme = useTheme()
      const reverse = ref(false)
      const icon = ref('$close')
      const rtl = ref(false)
      const truncate = ref('end')
      const page = ref(1)

      return { show, current: theme.current, reverse, icon, rtl, truncate, page }
    },
  }
</script>


```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by glen-84** [comment]
What do you think about using [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) to format the page numbers? (which would resolve #10085)

It would probably make sense to construct a single `NumberFormat` object somewhere (perhaps in the Lang/Locale service/composable), using the currently-set locale, and make that (or a proxy) available to components.

It's not only about larger numbers with separators, using a locale like `ar-EG` would actually render different numeric characters (f.e. `123` -> `١٢٣`).

**Comment by nekosaur** [comment]
@glen-84 It is a good point about locales like arabic. Personally I think it looks very weird to show `1,000` (or in the case of my own locale, `1 000`) in a pagination component, but it probably should be the default behavior. With maybe a prop to disable the formatting?

**Comment by glen-84** [comment]
I agree that it should be the default, as it matches the default of `Intl.NumberFormat`.

You could consider an option that effectively sets `useGrouping` to `false`, if there is demand for that. The naming might be a bit awkward though ... something like `number-grouping` may sound as if it would group the buttons in some way, and I assume that you'd want to keep the prop names fairly short.

You could bikeshed on the naming, or just defer adding the option until someone actually asks for it. 🙃

**Comment by glen-84** [comment]
I guess #10085 can be added to the `fixes` list? 🙂

File: packages/vuetify/src/components/VBtn/VBtn.tsx [hunk 1]
```diff
@@ -40,7 +40,10 @@ export default defineComponent({
     block: Boolean,
     stacked: Boolean,
 
-    color: String,
+    color: {
```
Comment (johnleider) [inline packages/vuetify/src/components/VBtn/VBtn.tsx:43]: What is this for?

File: packages/vuetify/src/components/VPagination/VPagination.tsx [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+    },
+    nextAriaLabel: {
+      type: String,
+      default: '$vuetify.pagination.ariaLabel.next',
+    },
+    lastAriaLabel: {
+      type: String,
+      default: '$vuetify.pagination.ariaLabel.last',
+    },
+    color: {
```
Comment (johnleider) [inline packages/vuetify/src/components/VPagination/VPagination.tsx:91]: Remove default color.

**Review by johnleider — CHANGES_REQUESTED** [review]
SASS structure needs updating and removal of default colors.

**Comment by nekosaur** [comment]
> removal of default colors.

I think we either need to have a discussion on how you think this should be done, or you can just push changes yourself.

**Comment by johnleider** [comment]
I started working on a branch testing the removal of v-btn and some other minor refactoring. 
`feat/v3-pagination-remove-v-btn`

**Comment by johnleider** [comment]
Requesting **Final** comments from @vuetifyjs/core-team

**Review by johnleider — APPROVED** [review]

**Review by KaelWD — APPROVED** [review]

**Pull request merged**